akka-stream/src/main/scala/DocumentStage.scala

```diff
@@ -150,8 +167,14 @@ private[akkastream] class DocumentStage[T](
 
         case _ => {
           last = None
-          Thread.sleep(1000) // TODO
-          onPull()
+          materializer.scheduleOnce(
```

Comment:
Either way that's a workaround, so I would keep it simple until that's really fixed

Reply:
reverted to previous code.